src/gluonnlp/cli/data/music_generation/prepare_music_midi.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import os
+import tarfile
+from gluonnlp.base import get_data_home_dir
+from gluonnlp.utils.misc import download
+import zipfile
+
+_CITATIONS = """
+@phdthesis{raffel2016learning,
+  title={Learning-based methods for comparing sequences, with applications to audio-to-midi alignment and matching},
+  author={Raffel, Colin},
+  year={2016},
+  school={Columbia University}
+}
+
+@inproceedings{hawthorne2018enabling,
+  title={Enabling Factorized Piano Music Modeling and Generation with the {MAESTRO} Dataset},
+  author={Curtis Hawthorne and Andriy Stasyuk and Adam Roberts and Ian Simon and Cheng-Zhi Anna Huang and Sander Dieleman and Erich Elsen and Jesse Engel and Douglas Eck},
+  booktitle={International Conference on Learning Representations},
+  year={2019},
+  url={https://openreview.net/forum?id=r1lYRjC9F7},
+}
+"""
+
+
+_CURR_DIR = os.path.realpath(os.path.dirname(os.path.realpath(__file__)))
+_BASE_DATASET_PATH = os.path.join(get_data_home_dir(), 'music_midi_data')
+
+_URL_FILE_STATS_PATH = os.path.join(_CURR_DIR, '..', 'url_checksums', 'music_midi.txt')
+_URL_FILE_STATS = dict()
+for line in open(_URL_FILE_STATS_PATH, 'r', encoding='utf-8'):
+    url, hex_hash, file_size = line.strip().split()
+    _URL_FILE_STATS[url] = hex_hash
+
+
+_URLS = {
+    'lmd_full': 'http://hog.ee.columbia.edu/craffel/lmd/lmd_full.tar.gz',
+    'lmd_matched': 'http://hog.ee.columbia.edu/craffel/lmd/lmd_matched.tar.gz',
+    'lmd_aligned': 'http://hog.ee.columbia.edu/craffel/lmd/lmd_aligned.tar.gz',
+    'clean_midi': 'http://hog.ee.columbia.edu/craffel/lmd/clean_midi.tar.gz',
+    'maestro_v1': 'https://storage.googleapis.com/magentadata/datasets/maestro/v1.0.0/maestro-v1.0.0-midi.zip',
+    'maestro_v2': 'https://storage.googleapis.com/magentadata/datasets/maestro/v2.0.0/maestro-v2.0.0-midi.zip',
+    'geocities': 'https://archive.org/download/archiveteam-geocities-midi-collection-2009/2009.GeoCities.MIDI.ArchiveTeam.zip'
+}
```

Comment:
@liangliannie @aashiqmuhamed @GomuGomuGo I've included three datasets that are public: different types of LakhMIDI, Maestro and Geocities.